File: Daily Exercises/Mar 2021/2021-03-30 : 74. Search a 2D Matrix/74. Search a 2D Matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m=matrix.length,n=matrix[0].length;
        int left=0,right=m*n-1;
        while(left<=right){
            int mid=(left+right)/2;
            int row=mid/n;
            int col=mid%n;
            if(matrix[row][col]<target){
                left=mid+1;
            }else if(matrix[row][col]>target){
                right=mid-1;
            }else{
                return true;
            }
        }
        return false;
    }
}

File: Daily Exercises/Mar 2021/2021-03-30 : 74. Search a 2D Matrix/74. Search a 2D Matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m,n=len(matrix),len(matrix[0])
        left=0
        right=m*n-1
        while(left<=right):
            mid=(left+right)//2
            row=mid//n
            col=mid-row*n
            if matrix[row][col]<target:
                left=mid+1
            elif matrix[row][col]>target:
                right=mid-1
            else:
                return True
        return False
